EncoreTickets.SDK.Tests/IntegrationTests/BasketServiceForTodayTixTests.cs

```diff
@@ -0,0 +1,605 @@
+        {
+            var reference = "invalid";
+
+            var exception = Assert.Catch<ApiException>(() =>
+            {
+                var result = service.GetBasketDetails(reference);
+            });
+
+            AssertApiException(exception, HttpStatusCode.NotFound);
+        }
+
+        [Test]
+        public void GetBasketDetails_IfBasketNotFound_Exception404()
+        {
+            var reference = configuration["Basket:TestBasketReferenceNotFound"];
+
+            var exception = Assert.Catch<ApiException>(() =>
+            {
+                var result = service.GetBasketDetails(reference);
+            });
+
+            AssertApiException(exception, HttpStatusCode.NotFound);
+        }
+
+        #endregion
+
+        #region GetBasketDeliveryOptions
+
+        [Test]
+        public void GetBasketDeliveryOptions_Successful()
+        {
+            var reference = configuration["Basket:TestBasketReference"];
+
+            var result = service.GetBasketDeliveryOptions(reference);
+
+            Assert.NotNull(result);
+            foreach (var deliveryOption in result)
+            {
+                Assert.NotNull(deliveryOption.Charge);
+                Assert.NotNull(deliveryOption.Method);
+            }
+        }
+
+        [Test]
+        public void GetBasketDeliveryOptions_IfBasketReferenceIsInvalid_Exception404()
+        {
+            var reference = "invalid";
+
+            var exception = Assert.Catch<ApiException>(() =>
+            {
+                var result = service.GetBasketDetails(reference);
+            });
+
+            AssertApiException(exception, HttpStatusCode.NotFound);
+        }
+
+        [Test]
+        public void GetBasketDeliveryOptions_IfBasketNotFound_Exception404()
+        {
+            var reference = configuration["Basket:TestBasketReferenceNotFound"];
+
+            var exception = Assert.Catch<ApiException>(() =>
+            {
+                var result = service.GetBasketDetails(reference);
+            });
+
+            AssertApiException(exception, HttpStatusCode.NotFound);
+        }
+
+        #endregion
+
+        #region UpsertBasket
+
+        [Test]
+        public void UpsertBasket_GetBasket_IfNewBasketAndBasketIsUsedAndWithoutFlexi_Successful()
+        {
+            var upsertBasketResult = (Basket.Models.Basket)null;
+            try
+            {
+                var reference = configuration["Basket:TestReferences:0"];
+                var request = CreateDefaultBasket(reference);
+                upsertBasketResult = service.UpsertBasket(request);
+                const int expectedReservationsCount = 1;
+
+                var basketDetails = service.GetBasketDetails(upsertBasketResult.Reference);
+
+                AssertUpsertBasketSuccess(request, basketDetails, expectedReservationsCount);
+                Assert.AreEqual(upsertBasketResult.Reference, basketDetails.Reference);
```

Comment:
No, because the fields of the returned basket can be different, just like for responses from basket api v1. And also changes to basket api v2 are in progress, so we need to update SDK later when it becomes clear that basket is being created as expected